File: backend/Services/Parsers/WaffleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class WaffleParser : BasicScoreResultParser
    {
        private ILogger<WaffleParser> _logger;

        public WaffleParser(ILogger<WaffleParser> logger) : base(logger)
        {
            _logger = logger;
        }
        public override string GameName => "Waffle";
        public override bool GolfScoring => false;
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"#{GameName.ToLower()}\\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => "wafflegame.net";
        public override string Url => "https://wafflegame.net/";
    }
}

File: lambdas/src/parsers/waffle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class WaffleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Waffle';
  readonly golfScoring = false;
  readonly helpText = null;
  protected readonly extraContent = 'wafflegame.net';
  readonly url = 'https://wafflegame.net/';
  
  // Matches "#waffle1395 X/5" or "#waffle113 0/5"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `#${this.gameName.toLowerCase()}\\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
